django/core/management/commands/migrate.py

```diff
@@ -156,6 +160,52 @@ def handle(self, *args, **options):
         else:
             targets = executor.loader.graph.leaf_nodes()
 
+        if options['prune']:
+            if not options['app_label']:
+                raise CommandError(
+                    'Migrations can be pruned only when an app is specified.'
+                )
+            if self.verbosity > 0:
+                self.stdout.write('Pruning migrations:', self.style.MIGRATE_HEADING)
+            to_prune = set(executor.loader.applied_migrations) - set(executor.loader.disk_migrations)
+            squashed_migrations_with_deleted_replaced_migrations = [
+                migration_key
+                for migration_key, migration_obj in executor.loader.replacements.items()
+                if any(replaced in to_prune for replaced in migration_obj.replaces)
+            ]
+            if squashed_migrations_with_deleted_replaced_migrations:
+                self.stdout.write(self.style.NOTICE(
+                    "  Cannot use --prune because the following squashed "
+                    "migrations are not recorded as applied and have their "
+                    "'replaces' attributes:"
+                ))
+                for migration in squashed_migrations_with_deleted_replaced_migrations:
+                    app, name = migration
+                    self.stdout.write(f'    {app}.{name}')
+                self.stdout.write(self.style.NOTICE(
+                    "  Re-run 'manage.py migrate' to record them as applied, "
+                    "and remove 'replaces' attributes in their Migration "
+                    "classes."
+                ))
```

Comment:
The previous phrasing tried to imply that re-running `manage.py migrate` was only if need be, but this phrasing now suggests that everything printed to the terminal is incompletely applied, which isn't the case: this is printing anything that has deleted migrations in `replaces`.

Should we go back to the prior wording or add another check to find out if the squashed migration was incompletely applied and have a specific warning for that here?

Reply:
Much better! (I agree the prior version needed improvement)

Reply:
I'll push a proposed edit. I think the first sentence still needs adjusting, too--it's more like and/or, not and.

Reply:
The second part about `replaces` is always true. I'm not sure if adding `and/or` will make it clearer :thinking:, maybe
```
  Cannot use --prune because the following squashed migrations have their 'replaces' attributes and may not be recorded as applied and 
    migrations.0001_squashed_0002
  Re-run 'manage.py migrate' if they are not marked as applied, and remove 'replaces' attributes in their Migration classes.
```

Reply:
Think that's it! 🚀